misaligned.cpp

```diff
@@ -1,21 +1,40 @@
 #include <assert.h>
 #include <iostream>
+#include <string>
+#include <sstream>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for (i = 0; i < 5; i++) {
-        for (j = 0; j < 5; j++) {
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+std::string printColorMap() {
+    std::stringstream ss;
+    std::cout << "printed output : "<< "\n";
+    for (int i = 0; i < 5; i++) {
+        for (int j = 0; j < 5; j++) {
+            ss << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
             std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
-    return i * j;
+    return ss.str();
+}
+
+void testprintColorMap() {
+    // Generate expected output dynamically and compare with
+    std::string printed_output = printColorMap();
+    std::stringstream expectedOutput;
+    int p_Num = 1;
+    std::cout << "expected output : "<< "\n";
+    for (int i = 0; i < 5; i++) {
+        for (int j = 0; j < 5; j++) {
+            std::cout << p_Num << " | " << majorColor[i] << " | " << minorColor[j] << "\n";
+            expectedOutput << p_Num++ << " | " << majorColor[i] << " | " << minorColor[j] << "\n";

```

Comment:
Good, in the expected output, you have captured the fact that pair numbers must start with 1.
However, the column separators are misaligned in the expected output.
Try to strengthen the test, so that "passing" gives confidence that the output will be "acceptable" to the user.